cs/client.py

```diff
@@ -266,15 +259,48 @@ def read_config(ini_group=None):
             ", ".join(paths)))
     conf = ConfigParser()
     conf.read(paths)
-    try:
-        cs_conf = conf[ini_group]
-    except AttributeError:  # python 2
-        cs_conf = dict(conf.items(ini_group))
-    cs_conf['name'] = ini_group
 
-    allowed_keys = ('endpoint', 'key', 'secret', 'timeout', 'method', 'verify',
-                    'cert', 'name', 'retry', 'theme', 'expiration')
+    if not ini_group:
+        ini_group = os.getenv('CLOUDSTACK_REGION', 'cloudstack')
+
+    if not conf.has_section(ini_group):
+        return dict(name=None)
+
+    ini_config = {k: v
+                  for k, v in conf.items(ini_group)
+                  if k in REQUIRED_CONFIG_KEYS + ALLOWED_CONFIG_KEYS}
```

Comment:
Switching to sets would mandate using `.union()` instead of + here and in `read_config`

```suggestion
                  if k in REQUIRED_CONFIG_KEYS.union(ALLOWED_CONFIG_KEYS)}
```